comp/logs/agent/agentimpl/agent.go

```diff
@@ -240,9 +237,31 @@ func (a *logAgent) startPipeline() {
 		a.pipelineProvider,
 		a.diagnosticMessageReceiver,
 		a.launchers,
-		a.schedulers,
 	)
 	starter.Start()
+
+	if !sds.ShouldBlockCollectionUntilSDSConfiguration(a.config) {
+		a.startSchedulers()
+		a.log.Info("logs-agent started")
+
+		a.started.Store(status.StatusRunning)
+	} else {
+		a.log.Info("logs-agent ready, schedulers not started: waiting for an SDS configuration to start the logs collection")
+		a.started.Store(status.StatusCollectionNotStarted)
+	}
+}
+
+func (a *logAgent) startSchedulers() {
+	a.prepareSchedulers.Do(func() {
+		a.schedulers.Start()
+
+		for _, scheduler := range a.schedulerProviders {
+			a.AddScheduler(scheduler)
+		}
+
+		a.log.Info("logs-agent schedulers started")
```

Comment:
This should probably say `logs-agent started` instead. The important thing for the person reading the logs is that after this point the logs agent is fully operational and will forward the messages. They may not be fluent with internals enough (users, new support engineers) to understand that schedulers was the last bit that wasn't ready. This way, SDS or not, when the logs agent starts forwarding, it will log `logs-agent started`.

Reply:
While I see the point, the `logs-agent started` still appears once the logs agent is fully started, the user experience stays the same relative to that.
E.g. without waiting for a configuration:
```
2024-07-24 14:50:47 CEST | CORE | INFO | (agent.go:262 in func1) | logs-agent schedulers started
2024-07-24 14:50:47 CEST | CORE | INFO | (agent.go:245 in startPipeline) | logs-agent started
```

When waiting for a configuration:
```
2024-07-24 14:50:47 CEST | CORE | INFO | (agent.go:262 in func1) | logs-agent schedulers started
... random other logs ...
2024-07-24 14:50:47 CEST | CORE | INFO | (agent.go:245 in startPipeline) | logs-agent started
```

I don't think the UX is worse here, except we have more information on the starting process of the logs agent.

Reply:
We are on the same page, your examples are exactly what I think should happen. But unless I'm missing something, `startPipeline` only logs `logs-agent started` in non-waiting code path, and it isn't called at all after applying SDS configuration, so the second example seems impossible with the current code?

Reply:
You are right.. address both your comments (logs & storing in the atomic only once)